api/core/v1alpha1/cnset_types.go

```diff
@@ -15,12 +16,62 @@ type CNSetSpec struct {
 type CNSetBasic struct {
 	PodSet `json:",inline"`
 
+	// CloneSetCommon kruise clone set common config for cn set
+	// +optional
+	CloneSetCommon `json:",inline"`
+
+	// ScaleStrategy clone set scalestrategy config for cn service
+	// +optional
+	ScaleStrategy CloneSetScaleStrategy `json:"scaleStrategy,omitempty"`
+
+	// Updatestrategy clone set updatestrategy config for cn service
+	// +optional
+	UpdateStrategy CloneSetUpdateStrategy `json:"updateStrategy,omitempty"`
+
+	InitialConfig CNInitialConfig `json:"initialConfig,omitempty"`
+
+	// ConfigMap is reference to a key in a config map
+	// +optional
+	ConfigMap *corev1.ConfigMapKeySelector `json:"configmap,omitempty"`
+
+	// ServiceType is the service type of cn service
+	// +optional
+	// +kubebuilder:default=ClusterIP
+	// +kubebuilder:validation:Enum=ClusterIP;NodePort;LoadBalancer
+	ServiceType corev1.ServiceType `json:"serviceType,omitempty"`
+
 	// CacheVolume is the desired local cache volume for CNSet,
 	// node storage will be used if not specified
 	// +optional
 	CacheVolume *Volume `json:"cacheVolume,omitempty"`
 }
 
+type CNInitialConfig struct {
+	PipelineConfig `json:",inline"`
+}
```

Comment:
I recommend not specify static type of config, since the configuration of CN will evolve in new versions, use `TomlConfig` would be more general